### PR TITLE
Remove trx_test_gen_plugin

### PR DESCRIPTION
**Pull request by oschwaldp-oci**
#### Test Updates
Tests previously using txn_test_gen_plugin have been updated to use the txn_generator.  Included tests:
- `nodeos_contrl_c_test.py`
- `nodeos_startup_catchup.py`
- `nodeos_snapshot_diff_tests.py`

#### Transaction Generator Updates (and supporting scripts)

- `peer-endpoint` & `port` - Support for defining the p2p peer endpoint port was added to the transaction generator and transaction generators launcher to allow directing generated transactions to a specific node/peer.
- `stop-on-trx-failed` - Added support for configuring generators to stop and clean up if transactions fail to send.
- Created `performance_tests` python module to make it easier to utilize transaction generator launcher from other tests.
- Added `waitToComplete` argument to the `TransactionGeneratorsLauncher`'s `launch` function to provide the ability to not wait on transactions generators to run to completion.

#### Cluster.py Update

- Adds `p2pBasePort` and means to calculate a given node's p2p listen port from node id (`getNodeP2pPort`).

### CODE REVIEW

File: tests/nodeos_startup_catchup.py [hunk 1]
```diff
@@ -102,23 +108,34 @@ def waitForNodeStarted(node):
...
+    testTrxGenDurationSec=60*60
+
+    tpsTrxGensConfig = TpsTrxGensConfig(targetTps=targetTps, tpsLimitPerGenerator=tpsLimitPerGenerator)
+    trxGenLauncher = TransactionGeneratorsLauncher(chainId=chainId, lastIrreversibleBlockId=lib_id,
+                                                   handlerAcct=cluster.eosioAccount.name, accts=f"{account1Name},{account2Name}",
+                                                   privateKeys=f"{account1PrivKey},{account2PrivKey}", trxGenDurationSec=testTrxGenDurationSec,
+                                                   logDir=Utils.DataDir, peerEndpoint=node0.host, port=producerP2pPort, tpsTrxGensConfig=tpsTrxGensConfig)
+
+    Print("Launch txn generators and start generating/sending transactions")
+    trxGenLauncher.launch(waitToComplete=False)
```
Comment (heifner) [inline tests/nodeos_startup_catchup.py:134]: It looks like this boiler-plate code is pretty much the same except for a few params. Can it all be wrapped into a `launchTrxGenerator` method of `Cluster` or `Node`?

File: tests/trx_generator/main.cpp [hunk 2]
```diff
@@ -95,6 +101,11 @@ int main(int argc, char** argv) {
          return INITIALIZE_FAIL;
       }
 
+      if(vmap.count("peer-endpoint") && peer_endpoint == "localhost") {
+         ilog("Initializing peer-endpoint from localhost to 127.0.0.1");
+         peer_endpoint = "127.0.0.1";
+      }
```
Comment (heifner) [inline tests/trx_generator/main.cpp:107]: Curious why this is needed?

**Review by heifner — CHANGES_REQUESTED** [review]

File: tests/trx_generator/main.cpp (hunk 2 above)
Comment (spoonincode) [reply tests/trx_generator/main.cpp:107]: I have a guess: `localhost` resolving to `::1` but p2p plugin only listens on IPv4. If I'm right, could consider changing trx_provider.cpp to resolve only IPv4 addresses.

Comment (heifner) [reply tests/trx_generator/main.cpp:107]: That make sense. You will need something like this in `p2p_connection::connect()`:
```
      tcp::resolver r(_p2p_service);
      tcp::resolver::query q(tcp::v4(), _peer_endpoint, std::to_string(_peer_port));
      auto i = r.resolve(q);
      boost::asio::connect(_p2p_socket, i);
```

Comment (oschwaldp-oci) [reply tests/trx_generator/main.cpp]: Thanks, both of you.  That was helpful.  @heifner, that worked a charm. Addressed: [6253abc](https://github.com/AntelopeIO/leap/pull/634/commits/6253abcc4958d1dc5757411fa748bd14b94e8d2d)

File: tests/nodeos_startup_catchup.py (hunk 1 above)
Comment (oschwaldp-oci) [reply tests/nodeos_startup_catchup.py]: I believe this addresses your suggestion: [1842807](https://github.com/AntelopeIO/leap/pull/634/commits/18428070e9bfe91c602965e2253cf733e15c0ecd)

**Review by spoonincode — APPROVED** [review]

**Review by heifner — APPROVED** [review]

**Pull request merged**